engine/apps/google/client.py

```diff
@@ -0,0 +1,67 @@
+import datetime
+import logging
+import typing
+
+from django.conf import settings
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+
+from apps.google.types import GoogleCalendarEvent
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+class GoogleCalendarAPIClient:
+    MAX_NUMBER_OF_CALENDAR_EVENTS_TO_FETCH = 250
+    """
+    By default the value is 250 events. The page size can never be larger than 2500 events
+    """
+
+    def __init__(self, access_token: str, refresh_token: str):
+        """
+        https://developers.google.com/calendar/api/quickstart/python
+        https://google-auth.readthedocs.io/en/stable/reference/google.oauth2.credentials.html
+        """
+        credentials = Credentials(
+            token=access_token,
+            refresh_token=refresh_token,
+            token_uri="https://www.googleapis.com/oauth2/v3/token",
+            client_id=settings.SOCIAL_AUTH_GOOGLE_OAUTH2_KEY,
+            client_secret=settings.SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET,
+        )
+
+        self.service = build("calendar", "v3", credentials=credentials)
+
+    def fetch_out_of_office_events(self) -> typing.List[GoogleCalendarEvent]:
+        """
+        https://developers.google.com/calendar/api/v3/reference/events/list
+        """
+
+        def _format_datetime_arg(dt: datetime.datetime) -> str:
+            """
+            https://stackoverflow.com/a/17159470/3902555
+            """
+            return dt.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+        now = _format_datetime_arg(datetime.datetime.now(datetime.UTC))
+
+        logger.info(
+            f"GoogleCalendarAPIClient - Getting the upcoming {self.MAX_NUMBER_OF_CALENDAR_EVENTS_TO_FETCH} "
+            "out of office events"
+        )
+
+        events_result = (
+            self.service.events()
+            .list(
+                calendarId="primary",
```

Comment:
This is fixed to fetch events from the `primary` calendar only, correct? Maybe define this as a class value? Should this eventually be configurable?

Reply:
correct 👍 good idea, moved to a class value + added some notes on what it would take to allow the user telling us which calendar they want to consider